File: MD.CMS.Administration/MD.CMS.Administration.Core.Web/scripts/businessLogic/dataAccess/controllers/options/v2/iContentRequestOptions.ts
/// <reference path="./enums/contentEnum.ts" />

namespace mdBusinessLogic.dataAccess.controllers.options.v2 {
    export interface iContentRequestOptions extends iSortableRequestOptions<enums.contentEnum>, iPageableRequestOptions, iSearchableRequestOptions {
        ContentIds?: Array<string>;
        OnlyPublished?: boolean;
        LoadAuthor?: boolean;
        FillFields?: boolean;
        FillMetaData?: boolean;
        Lcid?: number;
        FolderId?: number;
        TaxonomyId?: number;
        MenuId?: number;
        Alias?: string;
        DataBound?: boolean;
        ContentTypeId?: number;
    }
}

File: MD.CMS.BusinessLogic.Core/DataAccess/Controllers/V2/Options/Interfaces/IContentRequestOptions.cs
using System.Collections.Generic;

namespace MD.CMS.BusinessLogic.Core.DataAccess.Controllers.V2.Options.Interfaces
{
    public interface IContentRequestOptions : IPageableRequestOptions, ISearchableRequestOptions, ISortableRequestOptions<Enumerations.Mapping.ContentEnum>
    {
        IEnumerable<string> ContentIds { get; }
        bool OnlyPublished { get; set; }
        bool LoadAuthor { get; }
        bool FillFields { get; }
        bool FillMetaData { get; }
        int Lcid { get; }
        long FolderId { get; }
        long TaxonomyId { get; }
        long MenuId { get; }
        string Alias { get; }
        bool DataBound { get; }
        long ContentTypeId { get; }
    }
}
